### PR TITLE
Version 1.2.0

### LINKED ISSUE #4211: Make a new OSCAR release?

### PR DESCRIPTION
**Pull request by fingolfin**
Resolves #4211

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "Oscar"
 uuid = "f1435218-dba5-11e9-1e4d-f1a5fab5fc13"
 authors = ["The OSCAR Team <oscar@oscar-system.org>"]
-version = "1.2.0-DEV"
+version = "0.12.0"
```
Comment (benlorenz) [inline Project.toml]:
```suggestion
version = "1.2.0"
```
Can we make that 1.2.0 instead? (including the other places, probably running the script again)

**Comment by lgoettgens** [comment]
I think you messed up `1.2.0` with `0.12.0`.

And just as stupid question since I don't remember how we did it the last few times: When do we branch off `release-1.2`? Before or after adapting the version?

**Comment by benlorenz** [comment]
> And just as stupid question since I don't remember how we did it the last few times: When do we branch off `release-1.2`? Before or after adapting the version?

I usually created the release branch first and then merged the PR setting the version to the release version. But I guess bumping it first and then branching from the merge is also fine.
Once the branch is done the master should get 1.3.0-DEV.

I wrote down my process here: https://docs.oscar-system.org/stable/DeveloperDocumentation/release/

File: Project.toml (hunk 1 above)
Comment (fingolfin) [reply Project.toml]: d'oh. of course. Argh

**Comment by fingolfin** [comment]
Yeah my plan is that as soon as as this is merged, we merge another PR setting the version to 1.3.0-DEV.

Or I could just put that commit in here right now and we "merge by rebase" (instead of squash or actual merge) to preserve both commits.

**Comment by thofma** [comment]
Thanks. I was not sure about the process with the branches (which is why I did not open the PR myself).

**Comment by fingolfin** [comment]
@benlorenz I assume you will create the branch, ok?

**Comment by benlorenz** [comment]
> @benlorenz I assume you will create the branch, ok?

done: https://github.com/oscar-system/Oscar.jl/commits/release-1.2/
_Edit:_ I also added branch protection rules.

**Comment by lgoettgens** [comment]
@benlorenz can you do the 1.3.0-DEV bump as well an commit it as the next thing on master?

**Comment by benlorenz** [comment]
> @benlorenz can you do the 1.3.0-DEV bump as well an commit it as the next thing on master?

#4258

**Comment by benlorenz** [comment]
I regenerated the release notes for 1.2.0 with github since tagbot only showed the changes since the when 1.1.2 was created.

**Pull request merged**